feat(#1943): migrating from 4.x medic/helm-charts to 5.x cht-core helm-charts (#1989)
medic/cht-docs: content/en/hosting/cht/migration/helm-charts-4x-to-5x-migration.md

# CHT 4.x to 5.x Helm Charts Migration Guide

## Overview

This guide covers the migration of Kubernetes based CHT 4.x deployments using the [legacy Helm charts](https://github.com/medic/helm-charts) to the [new charts](https://github.com/medic/cht-core/tree/master/scripts/build/helm) introduced in CHT 5.x. The new Helm chart require updates to your `values.yaml` file. 

Before starting be sure you have a `git clone` of the [CHT Core repository](https://github.com/medic/cht-core/) and that you have updated it with `git pull origin` to ensure you have the latest changes locally.  

## Migration Steps

{{% steps %}}

### Step 1: Prepare Migration Values File

1. Export the values from your existing 4.x deployment into the `migration-5x-values.yaml` file:
   ```bash
   helm get values <your-release-name> --namespace <your-namespace> > migration-5x-values.yaml
   ```

> 2. Update the `migration-5x-values.yaml` file to be compliant with  5.x deployment requirements:

Reviewer: ```suggestion
3. Update the `migration-5x-values.yaml` file to be compliant with  5.x deployment requirements:
```

   api:
     service:
       type: ClusterIP
   
   # Update image versions to 5.x
   chtversion: "5.0.0"  # or "master" for latest development
   cht_image_tag: "5.0.0"  # or "master" for latest development
   ```

   #### Platform-Specific Storage Configuration
   
{{< tabs items="GKE,EKS,K3s-K3d" >}}

{{< tab >}}
```yaml
couchdb:
  persistent_disk:
    size: "<size>"  # Set appropriate size for your needs
```
{{< /tab >}}

{{< tab >}}
```yaml
ebs:
  preExistingEBSVolumeSize: "<size>"  # Set appropriate size for your needs
```
{{< /tab >}}

{{< tab >}}
```yaml
couchdb:
  storage_class: "local-path"  # was: "local-storage"
```
{{< /tab >}}

{{< /tabs >}}

### Step 2: Perform the Migration

1. **Run the migration upgrade** (from any directory where you have your `migration-5x-values.yaml` file):
   ```bash
   helm upgrade <your-release-name> /path/to/cht-core/scripts/build/helm \
     -f migration-5x-values.yaml \
     --namespace <your-namespace>
   ```

2. **Monitor the upgrade**:
   ```bash
   helm status <your-release-name> --namespace <your-namespace>
   kubectl get pods -n <your-namespace>
   ```

### Step 3: Verify Migration Success

> 1. **Check pod status**:

Reviewer: ```suggestion
1. Check pod status:
```

   ```bash
   kubectl get pods -n <your-namespace>
   ```

> 2. **Verify new 5.x components**:

Reviewer: ```suggestion
2. Verify new 5.x components:
```

   ```bash
   # Look for the new couchdb-nouveau component
   kubectl get pods -n <your-namespace> | grep nouveau
   ```

> 3. **Test data integrity**
> 
> 4. **Test application access**

Reviewer: ```suggestion
3. Test data integrity

4. Test application access
```

# CHT 4.x to 5.x Helm Charts Migration Guide

> This guide covers the migration from CHT 4.x deployments using the legacy Helm charts in the [medic/helm-charts](https://github.com/medic/helm-charts) repository to the new charts introduced in CHT 5.x within the [medic/cht-core](https://github.com/medic/cht-core) repository.
> 
> ## Overview
> 
> The Community Health Toolkit (CHT) has moved its Helm charts from a separate repository (`medic/helm-charts`) to the main CHT Core repository (`medic/cht-core`).

Reviewer: ```suggestion
## Overview

This guide covers the migration from CHT 4.x deployments using the [legacy Helm charts](https://github.com/medic/helm-charts) to the [new charts](https://github.com/medic/cht-core) introduced in CHT 5.x. Helm charts were moved from the `helm-charts` [repository](https://github.com/medic/helm-charts/tree/main/charts/cht-chart-4x) to the main `cht-core` [repository](https://github.com/medic/cht-core/tree/master/scripts/build/helm).
```

### Key Differences

#### Chart Naming
- **Old**: `medic/cht-chart-4x`
- **New**: `cht-chart` (local chart from cht-core repository)

> ## Breaking Changes Summary

Reviewer: ```suggestion
## Breaking Changes
```

> **IMPORTANT**: The following breaking changes require updates to your existing `values.yaml` files.

Reviewer: ```suggestion
The following breaking changes require updates to your existing `values.yaml` files.
```

| **No default persistent volume sizes** | **CRITICAL** - Must set platform-specific storage sizes |

#### 4. Values File Structure Changes

| Change | Impact |
|--------|---------|
| **Hierarchical structure** | **MEDIUM** - May need to restructure flat values files |
| **Nested couchdb configuration** | **MEDIUM** - CouchDB settings now under `couchdb:` section |

### Platform-Specific Storage Changes

#### GKE (Google Kubernetes Engine)
- **Required**: `couchdb.persistent_disk.size` (no default)

#### EKS (Amazon Elastic Kubernetes Service)
- **Required**: `ebs.preExistingEBSVolumeSize` (no default)

### Minor Changes

#### Storage Class Changes
- **K3s-K3d Storage Class**: `local-storage` (4.x) → `local-path` (5.x) - **MINOR CHANGE**
  - **Action Required**: Update K3s-K3d storage class if using defaults

### Getting Help

If you encounter issues during migration:

1. Check the [CHT Documentation](https://docs.communityhealthtoolkit.org/)
2. Review the [CHT Forum](https://forum.communityhealthtoolkit.org/)
3. Open an issue in the [CHT Core Repository](https://github.com/medic/cht-core/issues)

## Migration Steps

> ### Step 1: Prepare Migration Values File

Reviewer: consider using [native hextra steps](https://imfing.github.io/hextra/versions/v0.10/docs/guide/shortcodes/steps/) instead of level 3 (`###`) titles.

   ```yaml
   couchdb:
     persistent_disk:
       size: "<size>"  # Set appropriate size for your needs
   ```
   
   **For EKS:**
   ```yaml
   ebs:
     preExistingEBSVolumeSize: "<size>"  # Set appropriate size for your needs
   ```
   
   **For K3s-K3d:**
   ```yaml
   couchdb:
     storage_class: "local-path"  # was: "local-storage"
   ```

### Step 2: Perform the Migration

1. **Navigate to your 4.x helm chart directory**:
   ```bash
   cd /path/to/your/4x/helm/charts
   ```

2. **Run the migration upgrade**:
   ```bash
   helm upgrade <your-release-name> /path/to/cht-core/scripts/build/helm \
     -f migration-5x-values.yaml \
     --namespace <your-namespace>
   ```

3. **Monitor the upgrade**:
   ```bash
   helm status <your-release-name> --namespace <your-namespace>
   kubectl get pods -n <your-namespace>
   ```

### Step 3: Verify Migration Success

1. **Check pod status**:
   ```bash
   kubectl get pods -n <your-namespace>
   ```

2. **Verify new 5.x components**:
   ```bash
   # Look for the new couchdb-nouveau component
   kubectl get pods -n <your-namespace> | grep nouveau
   ```

3. **Test data integrity**

4. **Test application access**

## Best Practices

> > **⚠️ WARNING**:

Reviewer: ```suggestion
```
